updater.go

```diff
@@ -57,11 +58,19 @@ func NewUpdater(source UpdateSource, config Config, log logging.Logger) *Updater
 }
 
 // Update checks, downloads and performs an update
-func (u *Updater) Update(ctx Context) (*Update, error) {
+func (u *Updater) Update(ctx Context) (*Update, *Error) {
```

Comment:
Cool, this will save you from ever needing an `*Error` type, which can be buggy because of the crazy `nil` v `nil` semantics in Go.